src/assign.js

```diff
@@ -0,0 +1,8 @@
+/**
+ * assign
+ * Non-destructively assign one object into another
+ */
+
+export default function assign (a, b) {
+  return { ...a, ...b }
```

Comment:
This spreads the properties of `a` and `b` into a new object. So it safely merges the two objects without mutating either of them.